src/Ilios/CoreBundle/Form/DataTransformers/ArrayToIdTransformer.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Ilios\CoreBundle\Form\DataTransformer;
+
+use Doctrine\ORM\EntityManager;
+use Symfony\Component\Form\DataTransformerInterface;
+
+class ArrayToIdTransformer implements DataTransformerInterface
+{
+    /**
+     * @var EntityManager
+     */
+    private $em;
+
+    /**
+     * @var string Name spaced entity
+     */
+    private $class;
+
+    /**
+     * @var array
+     */
+    private $metadata;
+
+    /**
+     * @param EntityManager $em
+     * @param string        $class
+     */
+    public function __construct(EntityManager $em, $class)
+    {
+        $this->em = $em;
+        $this->class = $class;
+    }
+
+    /**
+     * @param mixed $data
+     *
+     * @return mixed
+     */
+    public function transform($data)
+    {
+        return $data;
+    }
+
+    /**
+     * Transforms a string or array into an id.
+     */
+    public function reverseTransform($data)
+    {
+
+        if (!$data) {
+            return null;
+        }
+
+        if (is_scalar($data)) {
+            return $data;
+        }
+
+        //@todo lookup entity's identifier.  Assuming that "id" is the identifier
```

Comment:
Nitpick: It looks like this comment is wrong and we are correctly looking up the identifier.